cmake/modules/snippets.cmake

```diff
@@ -53,11 +53,20 @@ function(zephyr_process_snippets)
     file(REMOVE ${snippets_generated})
   else()
     string(REPLACE " " ";" SNIPPET_AS_LIST "${SNIPPET}")
+    if(Sysbuild_DIR)
+      string(REPLACE " " ";" SYSBUILD_SNIPPET_AS_LIST_LOCAL "${sysbuild_SNIPPET}")
+      list(APPEND SNIPPET_AS_LIST ${SYSBUILD_SNIPPET_AS_LIST_LOCAL})
+    endif()
     set(SNIPPET_AS_LIST "${SNIPPET_AS_LIST}" PARENT_SCOPE)
   endif()
 
   # Set SNIPPET_ROOT.
-  list(APPEND SNIPPET_ROOT ${APPLICATION_SOURCE_DIR})
+  zephyr_get(SNIPPET_ROOT MERGE SYSBUILD GLOBAL)
+  if(Sysbuild_DIR)
+    list(APPEND SNIPPET_ROOT ${APP_DIR})
+  else()
+    list(APPEND SNIPPET_ROOT ${APPLICATION_SOURCE_DIR})
+  endif()
```

Comment:
if we move `SNIPPET_ROOT` handling into the common root handling https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/root.cmake and similar for the sysbuild part https://github.com/zephyrproject-rtos/zephyr/blob/main/share/sysbuild/cmake/modules/sysbuild_root.cmake then it follows the same principle as other roots and we avoid sysbuild specifics in the common snippet module.

Reply:
Fixed